### PR TITLE
feat: do not close or manage lifecycle of http-client passed in

### PR DESCRIPTION
**Pull request by maxxedev**

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/java-http-client?pullRequest=25) <br/>All committers have signed the CLA.

File: src/main/java/com/sendgrid/Client.java [hunk 1]
```diff
@@ -48,13 +48,15 @@ public HttpDeleteWithBody(final String uri) {
 
 	private CloseableHttpClient httpClient;
 	private Boolean test;
+	private boolean createdHttpClient;
```
Comment (andy-trimble) [inline src/main/java/com/sendgrid/Client.java:51]: Set default value to `false`

**Review by andy-trimble — CHANGES_REQUESTED** [review]
Please add documentation to the `Client(CloseableHttpClient httpClient)` constructor indicating that the client will not be closed on finalization.

File: src/main/java/com/sendgrid/Client.java (hunk 1 above)
Comment (andy-trimble) [reply src/main/java/com/sendgrid/Client.java:51]: Disregard this comment. The default value will be false. That is sufficient.

**Review by andy-trimble — APPROVED** [review]

**Comment by andy-trimble** [comment]
LGTM

**Comment by maxxedev** [comment]
could this be merged please?

**Comment by thinkingserious** [comment]
@maxxedev,

LGTM

Please fix the conflict and I'll merge this :)

Thanks!

With Best Regards,

Elmer

**Comment by maxxedev** [comment]
ah conflicts. 
fixed now.

**Comment by thinkingserious** [comment]
Hello @maxxedev,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**